packages/bull-vault/src/BullStrategy.sol

```diff
@@ -157,6 +194,34 @@ contract BullStrategy is ERC20, LeverageBull {
         emit Withdraw(msg.sender, _bullAmount, wPowerPerpToRedeem);
     }
 
+    function shutdownRepayAndWithdraw(uint256 wethToUniswap, uint256 shareToUnwind) external {
+        require(msg.sender == shutdownContract, "BS4");
+        if (shareToUnwind == ONE) {
+            hasRedeemedInShutdown = true;
+        }
```

Comment:
So this is to execute the trade on many sizes? will those sizes execute simultaneously? do we need like a pausing state to activate during the unwind phase?

Reply:
We will need a pausing state if we remove have the isShutdown check on controller for deposit and withdrawal, but because the only time an unwind will work is if controller is shutdown, we don't need it, right now, at least from the perspective of public non-owner functions.